crypto/cmp/cmp_vfy.c

```diff
@@ -772,8 +772,16 @@ int ossl_cmp_msg_check_update(OSSL_CMP_CTX *ctx, const OSSL_CMP_MSG *msg,
 
     /* compare received nonce with the one we sent */
     if (!check_transactionID_or_nonce(ctx->senderNonce, hdr->recipNonce,
-                                      CMP_R_RECIPNONCE_UNMATCHED))
-        return 0;
+                                      CMP_R_RECIPNONCE_UNMATCHED)) {
+        /* compare received nonce with initial request sender nonce */
+        if (!(OSSL_CMP_MSG_get_bodytype(msg) != OSSL_CMP_PKIBODY_POLLREP
+              && ctx->reqsenderNonce != NULL
```

Comment:
No, at this point the `ctx->reqsenderNonce != NULL` has no real effect,
and you must, as before, return `0` in case the first `check_transactionID_or_nonce()` fails and `OSSL_CMP_MSG_get_bodytype(msg) != OSSL_CMP_PKIBODY_POLLREP || ctx->reqsenderNonce == NULL`

Reply:
It is updated as suggested latter.